compiler/rustc_trait_selection/src/solve/normalize.rs

```diff
@@ -41,15 +42,46 @@ pub fn deeply_normalize_with_skipped_universes<'tcx, T, E>(
     value: T,
     universes: Vec<Option<UniverseIndex>>,
 ) -> Result<T, Vec<E>>
+where
+    T: TypeFoldable<TyCtxt<'tcx>>,
+    E: FromSolverError<'tcx, NextSolverError<'tcx>>,
+{
+    let (value, goals) =
+        deeply_normalize_with_skipped_universes_and_ambiguous_goals(at, value, universes)?;
+    assert_eq!(goals, vec![]);
+
+    Ok(value)
+}
+
+/// Deeply normalize all aliases in `value`. This does not handle inference and expects
+/// its input to be already fully resolved.
+///
+/// Additionally takes a list of universes which represents the binders which have been
+/// entered before passing `value` to the function. This is currently needed for
+/// `normalize_erasing_regions`, which skips binders as it walks through a type.
+///
+/// TODO: doc
```

Comment:
I need to explain that this doesn't return *all* ambiguous preds, just the ones that are stalled on coroutines.